fhir-persistence-schema/docs/DB2MultiTenancy.md

```diff
@@ -1,16 +1,225 @@
+
+The tablespaces are created with automatic storage using an EXTENTSIZE of 4 (x32KB) pages to reduce overhead and improve deployment times. This is due to the large number of objects (tables and indexes) created in the schema, each requiring an initial number of extents. 
+
+To add a new tenant, a utility program connects to Db2 as the FHIRADMIN user and adds records to the FHIR_ADMIN.TENANTS and FHIR_ADMIN.TENANT_KEYS tables. The utility probram then uses the newly assigned mt_id value when adding new partitions to each of the tables in the FHIRDATA schema.
+
+The tablespace is managed automatically with an ([EXTENTSIZE](https://www.ibm.com/support/knowledgecenter/SSEPGG_11.5.0/com.ibm.db2.luw.admin.dbobj.doc/doc/c0004964.html)) of 128KB. Note, the extentsize is multiplied by the pagesize to arrive at a storage size.
+
+Each tenant is allocated a partition based on the MT_ID and assigned to a tablespace. The tenant `tablespace` is created one time and no changes to the tablespace are expected.
+
+In general, smaller extent sizes can harm the performance of table and index scans because they limit the amount of data which can be read in a single IO operation. However, such scans are not common in most FHIR query use-cases, so this tends to not be an issue.
+
+
+## Multi-tenant Queries
+
+We could use the multi-tenant enabled table definition as-is and modify every SQL statement to include `AND mt_id = ?` but this is error-prone and risky. If a query was missed or poorly written and ran without that predicate, it could expose data from another tenant.
+
+Instead, we leverage DB2 permission objects which allow us to specify a filter condition. Each query submitted by the application is rewritten by the database to include the `FOR ROWS WHERE` predicate, which is defined as: 
+
+```
+CREATE PERMISSION ROW_ACCESS ON ptng.patients
+         FOR ROWS WHERE patients.mt_id = ptng.session_tenant
+         ENFORCED FOR ALL ACCESS
+           ENABLE;
+
+ALTER TABLE ptng.patients ACTIVATE ROW ACCESS CONTROL;
+```
+
+The `session_tenant` above is defined as a session variable. The application must set the value to match the mt_id value configured for the tenant.
+
+Logically:
+
+```
+SET ptng.session_tenant = 1;
+SELECT * FROM ptng.patients;
+commit;
+```
+
+Internally, DB2 rewrites the query to look like this:
+
+```
+SELECT * FROM ptng.patients WHERE mt_id = 1;
+```
+
+Because the table `ptng.patients` is partitioned using mt_id, the query optimizer is able to perform a partition-prune, by limiting the query access to the partition associated with the tenant. This results in a query execution plan very similar in terms of performance to a single-tenant scenario without any partitioning.
```

Comment:
```suggestion
SET fhirdata.session_tenant = 1;
SELECT * FROM fhirdata.patients;
commit;
```

Internally, DB2 rewrites the query to look like this:

```
SELECT * FROM fhirdata.patients WHERE mt_id = 1;
```

Because the table `fhirdata.patients` is partitioned using mt_id, the query optimizer is able to perform a partition-prune, by limiting the query access to the partition associated with the tenant. This results in a query execution plan very similar in terms of performance to a single-tenant scenario without any partitioning.
```

Reply:
schema name is now "example"